solidity/contracts/BondedECDSAKeep.sol

```diff
@@ -527,127 +78,48 @@ contract BondedECDSAKeep is IBondedECDSAKeep {
         address _keepBonding,
         address payable _keepFactory
     ) public {
-        require(!isInitialized, "Contract already initialized");
+        initializeBondedECDSAKeep(
+            _owner,
+            _members,
+            _honestThreshold,
+            _keepBonding
+        );
 
-        owner = _owner;
-        members = _members;
-        honestThreshold = _honestThreshold;
         memberStake = _memberStake;
         tokenStaking = TokenStaking(_tokenStaking);
-        keepBonding = KeepBonding(_keepBonding);
         keepFactory = BondedECDSAKeepFactory(_keepFactory);
-        status = Status.Active;
-        isInitialized = true;
 
         tokenStaking.claimDelegatedAuthority(_keepFactory);
 
         for (uint256 i = 0; i < _members.length; i++) {
             tokenStaking.lockStake(_members[i], _stakeLockDuration);
         }
```

Comment:
Can we extract this code to `lockMemberStakes` function and place it next to `unlockMemberStakes`?